src/compiler.ts

```diff
@@ -6991,102 +7003,69 @@ export class Compiler extends DiagnosticEmitter {
-          let overloadSignature = overloadInstance.signature;
-          let overloadParameterTypes = overloadSignature.parameterTypes;
-          let overloadNumParameters = overloadParameterTypes.length;
-          let paramExprs = new Array<ExpressionRef>(1 + overloadNumParameters);
-          paramExprs[0] = module.local_get(0, sizeTypeRef); // this
-          for (let n = 1; n <= numParameters; ++n) {
-            paramExprs[n] = module.local_get(n, parameterTypes[n - 1].toRef());
-          }
-          let needsVarargsStub = false;
-          for (let n = numParameters; n < overloadNumParameters; ++n) {
-            // TODO: inline constant initializers and skip varargs stub
-            paramExprs[1 + n] = this.makeZero(overloadParameterTypes[n], overloadInstance.declaration);
-            needsVarargsStub = true;
-          }
-          let calledName = needsVarargsStub
-            ? this.ensureVarargsStub(overloadInstance).internalName
-            : overloadInstance.internalName;
-          let returnTypeRef = overloadSignature.returnType.toRef();
-          let stmts = new Array<ExpressionRef>();
-          if (needsVarargsStub) {
-            // Safe to prepend since paramExprs are local.get's
-            stmts.push(module.global_set(this.ensureArgumentsLength(), module.i32(numParameters)));
-          }
-          if (returnType == Type.void) {
-            stmts.push(
-              module.call(calledName, paramExprs, returnTypeRef)
-            );
-            stmts.push(
-              module.return()
-            );
-          } else {
-            stmts.push(
-              module.return(
-                module.call(calledName, paramExprs, returnTypeRef)
-              )
-            );
-          }
-          builder.addCase(classInstance.id, stmts);
-          // Also alias each extendee inheriting this exact overload
-          let extendees = classInstance.getAllExtendees(
-            isProperty
-              ? unboundOverloadParent.name
-              : instance.prototype.name
-          );
-          for (let _values = Set_values(extendees), a = 0, b = _values.length; a < b; ++a) {
-            let extendee = _values[a];
-            builder.addCase(extendee.id, stmts);
-          }
-        }
+        stmts.push(
+          module.return(
+            module.call(calledName, paramExprs, returnTypeRef)
+          )
+        );
+      }
+      let classInstance = assert(overloadInstance.getClassOrInterface());
+      builder.addCase(classInstance.id, stmts);
+      // Also alias each extendee inheriting this exact overload
+      let extendees = classInstance.getAllExtendees(
+        overloadInstance.isAccessor
+          // FIXME: Parent is not the property, but should be, so we don't need
+          // to use a substring here but can trivially use the property name.
+          ? instance.prototype.name.substring(GETTER_PREFIX.length)
```

Comment:
This is a little odd, in that resolving a getter or setter makes their instance reference the class as the parent (as natural when resolving methods), not the property instance (which is merely an ephemeral pre-resolve container). All we need here, however, is the name of the property, which we cannot trivially obtain due to the strange parent reference, so we end up with some unnecessary string garbage.